### PR TITLE
[FOCS] Cleanup species scripts

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
- Reorder macro uses
- Rename local variables to have leading underscore
- Rename local function to have leading underscore and lowercase
- Extract common code to functions

The diff mostly contains changes like:  

```diff
- Value - 1
+ Value + -1 
```

<details><summary>full diff</summary>
<p>

```diff
--- SP_CHATO.master
+++ SP_CHATO.sa-cleanup-species
@@ -1680,7 +1680,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_CHATO
             scope =
                 Source
@@ -1697,7 +1697,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_CHATO
             scope =
                 Source
@@ -1709,7 +1709,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_CHATO
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_CHATO
             scope =
                 Source
             activation =
--- SP_SLY.master
+++ SP_SLY.sa-cleanup-species
@@ -1559,7 +1559,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_SLY
             scope =
                 Source
@@ -1576,7 +1576,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_SLY
             scope =
                 Source
@@ -1588,7 +1588,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_SLY
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_SLY
             description = "GOOD_DETECTION_DESC"
             scope =
                 Source
--- SP_TAEGHIRUS.master
+++ SP_TAEGHIRUS.sa-cleanup-species
@@ -1666,7 +1666,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_TAEGHIRUS
             scope =
                 Source
@@ -1683,7 +1683,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_TAEGHIRUS
             scope =
                 Source
@@ -1695,7 +1695,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_TAEGHIRUS
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_TAEGHIRUS
             scope =
                 Source
             activation =
--- SP_EXOBOT.master
+++ SP_EXOBOT.sa-cleanup-species
@@ -1590,7 +1590,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_EXOBOT
             scope =
                 Source
@@ -1607,7 +1607,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_EXOBOT
             scope =
                 Source
@@ -1619,7 +1619,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_EXOBOT
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_EXOBOT
             scope =
                 Source
             activation =
@@ -1708,23 +1708,23 @@
                 ]
             priority = 100
             effects =
-                SetMaxShield value = Value - 0.500000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
+                SetMaxShield value = Value + -0.500000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
     Source
     DesignHasPart name = "SH_DEFENSE_GRID"
 ]
- - 1.000000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
+ + -1.000000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
     Source
     DesignHasPart name = "SH_DEFLECTOR"
 ]
- - 1.500000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
+ + -1.500000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
     Source
     DesignHasPart name = "SH_PLASMA"
 ]
- - 2.000000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
+ + -2.000000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
     Source
     DesignHasPart name = "SH_MULTISPEC"
 ]
- - 2.500000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
+ + -2.500000 * GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * Statistic If condition = And [
     Source
     DesignHasPart name = "SH_BLACK"
 ]
--- SP_DERTHREAN.master
+++ SP_DERTHREAN.sa-cleanup-species
@@ -1657,7 +1657,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_DERTHREAN
             scope =
                 Source
@@ -1674,7 +1674,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_DERTHREAN
             scope =
                 Source
@@ -1686,7 +1686,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_DERTHREAN
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_DERTHREAN
             scope =
                 Source
             activation =
--- SP_GYSACHE.master
+++ SP_GYSACHE.sa-cleanup-species
@@ -1685,7 +1685,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_GYSACHE
             scope =
                 Source
@@ -1702,7 +1702,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_GYSACHE
             scope =
                 Source
@@ -1714,7 +1714,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_GYSACHE
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_GYSACHE
             scope =
                 Source
             activation =
--- SP_SILEXIAN.master
+++ SP_SILEXIAN.sa-cleanup-species
@@ -1628,7 +1628,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 5.000000            ]
+                SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000                SetMaxCapacity partname = "SR_WEAPON_2_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -2.000000                SetMaxCapacity partname = "SR_WEAPON_3_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -3.000000                SetMaxCapacity partname = "SR_WEAPON_4_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -5.000000            ]
         EffectsGroup // from SP_SILEXIAN
             scope =
                 Source
@@ -1645,7 +1645,7 @@
                 ]
             priority = 100
             effects = [
-                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value - GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * 1.000000            ]
+                SetMaxSecondaryStat partname = "FT_HANGAR_2" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_3" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000                SetMaxSecondaryStat partname = "FT_HANGAR_4" value = Value + GameRule name = "RULE_FIGHTER_DAMAGE_FACTOR" * -1.000000            ]
         EffectsGroup // from SP_SILEXIAN
             scope =
                 Source
@@ -1657,7 +1657,7 @@
                 ]
             priority = 100
             effects =
-                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value - 1.000000        EffectsGroup // from SP_SILEXIAN
+                SetMaxSecondaryStat partname = "SR_WEAPON_0_1" value = Value + -1.000000        EffectsGroup // from SP_SILEXIAN
             scope =
                 Source
             activation =
[]
Finish comparing 50 files
Species parsed: 50
```

</p>
</details>

### CODE REVIEW

**Comment by geoffthemedio** [comment]
> ```diff
> - Value - 1
> + Value + -1 
> ```

why?

**Comment by Cjkjvfnby** [comment]
> Why?

There are two approaches: 

1. pass the bonus, with looks Pythonic, but produce this `+ -1`

```python
_ship_shields(tag="BAD", defence_grid=-0.5, deflector=-1.0, plasma=-1.5, multispec=-2.0, black=-2.5)
```
2. Pass the expression, which produce nice diff, but more complicated
```python
_fuel("BAD", Value - 0.5),
```

**Comment by geoffthemedio** [comment]
I don't understand your point... neither of those approaches look like `Value + -1`.

**Comment by Cjkjvfnby** [comment]
> I don't understand your point... neither of those approaches look like Value + -1.


We have a [function ](https://github.com/freeorion/freeorion/pull/4795/files#diff-44aaec5d2e64d75716393d8b96628f55a66e7df92458feae32dbf0757ed24999R5)that returns shared code :
```python
def _weapon(*, tag: str, tier_0: int, tier_1: int, tier_2: int, tier_3: int, tier_4: int, hangar: float):
    return [
        EffectsGroup(
            ...
            effects=[
                SetMaxDamage(partname="SR_WEAPON_1_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * tier_1),
````

When we [call it with ](https://github.com/freeorion/freeorion/pull/4795/files#diff-44aaec5d2e64d75716393d8b96628f55a66e7df92458feae32dbf0757ed24999R52)`tier_1 = 1`  

The result species dump will be the same as before:
```
SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000
```

When we [call it with ](https://github.com/freeorion/freeorion/pull/4795/files#diff-44aaec5d2e64d75716393d8b96628f55a66e7df92458feae32dbf0757ed24999R51) `tier_1 = -1`  

We will get this diff because in shared code we use + instead of -, [here](https://github.com/freeorion/freeorion/pull/4795/files#diff-44aaec5d2e64d75716393d8b96628f55a66e7df92458feae32dbf0757ed24999L17).

```diff
- SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value - GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * 1.000000
+ SetMaxCapacity partname = "SR_WEAPON_1_1" value = Value + GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR" * -1.000000       
```

Since `A - (B * C)` == `A + (B * -C)` is the same, the result will also be the same.

**Comment by o01eg** [comment]
I get what you mean, it's fine. But there is an error in the code:

```
2023-11-23T18:51:26.1917682Z   2: [2023-11-23 18:51:26.182463] [0x00001468] [info]    Traceback (most recent call last):
2023-11-23T18:51:26.1919304Z   2: [2023-11-23 18:51:26.182463] [0x00001468] [info]      File "<string>", line 79, in <module>
2023-11-23T18:51:26.1920208Z   2: [2023-11-23 18:51:26.182463] [0x00001468] [info]    TypeError: Value after * must be an iterable, not EffectsGroup
2023-11-23T18:51:26.1921449Z   2: [2023-11-23 18:51:26.182463] [0x00001468] [info]    Unable to parse data file D:\a\freeorion\freeorion\build\Release\test-scripting\species\SP_ABADDONI.focs.py
```

File: default/scripting/species/common/weapons.py [hunk 1]
```diff
@@ -1,152 +1,54 @@
...
-    EffectsGroup(
-        scope=IsSource,
-        activation=Ship & Armed & DesignHasPart(name="SR_WEAPON_0_1"),
-        effects=SetMaxSecondaryStat(partname="SR_WEAPON_0_1", value=Value + 3),
-    ),
-]
+    ]
+
+
+BAD_WEAPONS = _weapon(tag="BAD", tier_0=-1, tier_1=-1, tier_2=-2, tier_3=-3, tier_4=-5, hangar=-1.0)
```
Comment (agrrr3) [inline default/scripting/species/common/weapons.py:51]: as far as I can see/remember the cascading direct weapon damage steps (tier_*) are always equidistant (NB i was not aware, that death ray does two steps). so if you refactor, I think something akin to the following  signature would be better
```
BAD_WEAPONS = _weapon(tag="BAD", main_weapon_step=-1, fighter_damage_step=-1.0)
GOOD_WEAPONS = _weapon(tag="GOOD", main_weapon_step=1, fighter_damage_step=1.5)
GREAT_WEAPONS = _weapon(tag="GREAT", main_weapon_step=2, fighter_damage_step=3.0)
ULTIMATE_WEAPONS = _weapon(tag="ULTIMATE", main_weapon_step=3, fighter_damage_step=4.5)
```
(thinking more about this, actually it is the step size which gets set)

Comment (Cjkjvfnby) [reply default/scripting/species/common/weapons.py:51]: I'll make it as a separate request, this one is already too large.

**Comment by Cjkjvfnby** [comment]
> But there is an error in the code

Do we still need these tests?

**Comment by o01eg** [comment]
Yes, to test possible changes in future.

**Review by o01eg — APPROVED** [review]
Now looks fine.

**Pull request merged**